src/config/bookconfig.rs

```diff
@@ -5,7 +5,7 @@ use super::tomlconfig::TomlConfig;
 use super::jsonconfig::JsonConfig;
 
 /// Configuration struct containing all the configuration options available in mdBook.
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
```

Comment:
> That should probably be in a different commit than this one then, just to keep commit history clean. 

Good point. ATM this was just an exploratory PR for further review (and honestly I've removed the derives to check if this will sill compile). I'll cleanup if the PR will ever have a chance of merging.

> Course, I'm just a nobody, so take my recommendation with a grain of salt.

Nope. All help and another pair of eyes is always welcome. Thanks :+1: !